packages/plugin-hardhat/src/utils/deploy-impl.ts

```diff
@@ -116,17 +116,15 @@ async function deployImpl(
     },
     opts,
     undefined,
-    async (deploymentId: string, allowUndefined: boolean) => {
-      return await getDeploymentResponse(hre, deploymentId, allowUndefined);
-    },
+    remoteDeploymentId => getRemoteDeployment(hre, remoteDeploymentId),
   );
 
   let txResponse;
   if (opts.getTxResponse) {
     if ('deployTransaction' in deployment) {
       txResponse = deployment.deployTransaction;
     } else if (deployment.txHash !== undefined) {
-      txResponse = hre.ethers.provider.getTransaction(deployment.txHash);
+      txResponse = await hre.ethers.provider.getTransaction(deployment.txHash);
```

Comment:
This error was possible because this function returns `Promise<any>`. A cautionary tale! Is there any chance we could get rid of that `any`?